lib/shared.js

```diff
@@ -15,192 +32,104 @@ module.exports = {
     semi: "off",
-    "@typescript-eslint/semi": ["error", "always"],
+    "@typescript-eslint/semi": baseStyleRules.semi,
 
-    // Append `ts` and `tsx` to Airbnb 'import/extensions' rule
-    // Ensure consistent use of file extension within the import path
+    // Append 'ts' and 'tsx' to Airbnb 'import/extensions' rule
     // https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/extensions.md
     "import/extensions": [
-      "error",
-      "ignorePackages",
+      baseImportsRules["import/extensions"][0],
+      baseImportsRules["import/extensions"][1],
       {
-        js: "never",
-        mjs: "never",
-        jsx: "never",
+        ...baseImportsRules["import/extensions"][2],
         ts: "never",
         tsx: "never",
       },
     ],
 
-    // Append `ts` and `tsx` extensions to Airbnb 'import/no-extraneous-dependencies' rule
-    // Forbid the use of extraneous packages
+    // Append 'ts' and 'tsx' extensions to Airbnb 'import/no-extraneous-dependencies' rule
     // https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-extraneous-dependencies.md
-    // Paths are treated both as absolute paths, and relative to process.cwd()
     "import/no-extraneous-dependencies": [
-      "error",
+      baseImportsRules["import/no-extraneous-dependencies"][0],
       {
-        devDependencies: [
-          "test/**", // tape, common npm pattern
-          "tests/**", // also common npm pattern
-          "spec/**", // mocha, rspec-like pattern
-          "**/__tests__/**", // jest pattern
-          "**/__mocks__/**", // jest pattern
-          "test.{js,jsx,ts,tsx}", // repos with a single test file
-          "test-*.{js,jsx,ts,tsx}", // repos with multiple top-level test files
-          "**/*{.,_}{test,spec}.{js,jsx,ts,tsx}", // tests where the extension or filename suffix denotes that it is a test
-          "**/jest.config.js", // jest config
-          "**/vue.config.js", // vue-cli config
-          "**/webpack.config.js", // webpack config
-          "**/webpack.config.*.js", // webpack config
-          "**/rollup.config.js", // rollup config
-          "**/rollup.config.*.js", // rollup config
-          "**/gulpfile.js", // gulp config
-          "**/gulpfile.*.js", // gulp config
-          "**/Gruntfile{,.js}", // grunt config
-          "**/protractor.conf.js", // protractor config
-          "**/protractor.conf.*.js", // protractor config
-        ],
-        optionalDependencies: false,
+        ...baseImportsRules["import/no-extraneous-dependencies"][1],
+        devDependencies: baseImportsRules[
+          "import/no-extraneous-dependencies"
+        ][1].devDependencies.map(
+          glob =>
+            glob
+              .replace(/\.(jsx?)$/, ".{$1}") // Wrap file extensions in braces
+              .replace(/((\.*)js(x?))/g, "$1,$2ts$3"), // 'jsx?' => 'jsx?,tsx?'
+        ),
```

Comment:
This introduces unnecessary complexity to the regex matching for config files. For example, it will match `jest.config.js`, `jest.config.jsx`, `jest.config.ts`, and `jest.config.tsx`. But config files will only be available with `js` extension. I think the regex for config files should remain as-is.

Looking at the original settings from Airbnb:

```
devDependencies: [
        'test/**', // tape, common npm pattern
        'tests/**', // also common npm pattern
        'spec/**', // mocha, rspec-like pattern
        '**/__tests__/**', // jest pattern
        '**/__mocks__/**', // jest pattern
        'test.{js,jsx}', // repos with a single test file
        'test-*.{js,jsx}', // repos with multiple top-level test files
        '**/*{.,_}{test,spec}.{js,jsx}', // tests where the extension or filename suffix denotes that it is a test
        '**/jest.config.js', // jest config
        '**/jest.setup.js', // jest setup
        '**/vue.config.js', // vue-cli config
        '**/webpack.config.js', // webpack config
        '**/webpack.config.*.js', // webpack config
        '**/rollup.config.js', // rollup config
        '**/rollup.config.*.js', // rollup config
        '**/gulpfile.js', // gulp config
        '**/gulpfile.*.js', // gulp config
        '**/Gruntfile{,.js}', // grunt config
        '**/protractor.conf.js', // protractor config
        '**/protractor.conf.*.js', // protractor config
        '**/karma.conf.js' // karma config
      ],
```

It looks like, at this point in time, only items with `jsx` need to be expanded with `ts,tsx`. Perhaps we can only map those items here?